lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -362,6 +362,14 @@ spec = do
                             . slotStartTime sps
                     Just slot === f slot
 
+        it "slotStartTime (slotRange timeRange) ⊆ timeRange " $
+            withMaxSuccess 1000 $ property $
+                \sps timeRange -> do
+                    let r = getUniformTime <$> timeRange
+                    case fmap (slotStartTime sps) <$> slotRange sps r of
+                        Just r' -> r' `rangeIsSubrangeOf` r
+                        Nothing -> True
```

Comment:
I like where you're going!

I think a naive approach would lead to re-implementing `slotRange`, which would be undesirable.

There is a property very similar to (imagined) user-requirements for pagination I had imagined would be satisfied, but it is not!

If I have `r1 = [a, b]`, `r2 = [b,c]` I'd imagined that the union of calling `listTransactions` with `r1` and `r2` would be the same as calling `listTransactions` with `[a,c]` once.

I named `f = fmap slotStartTime . slotRange`, so more formally I'd expect something like `union $ map f (split timeRange n) === f timeRange`

This is actually an unlikely case since `f` shrinks ranges. We might get gaps, or we might even only get a few singleton ranges! 😱

<img width="914" alt="Skärmavbild 2019-08-22 kl  13 06 28" src="https://user-images.githubusercontent.com/304423/63513284-afeef380-c4dd-11e9-90ee-b3d3301e1c9d.png">

Reply:
So things went wrong in #595 already 😐 🔫

Reply:
No actually, this might as intended 🤔

Reply:
Tried two methods for a property, very messy still